### PR TITLE
[Merged by Bors] - feat: `Not` is involutive

### PR DESCRIPTION
**Pull request by IvanRenison**
---
For using it in #9131

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Logic/Lemmas.lean [hunk 1]
```diff
@@ -73,3 +73,7 @@ lemma Prop.exists {f : Prop → Prop} : (∃ p, f p) ↔ f True ∨ f False :=
   ⟨fun ⟨p, h⟩ ↦ by refine' (em p).imp _ _ <;> intro H <;> convert h <;> simp [H],
     by rintro (h | h) <;> exact ⟨_, h⟩⟩
 #align Prop.exists Prop.exists
+
+open Function
+
+lemma injective_not : Injective Not := fun _ _ ↦ by simpa using not_iff_not.mp
```
Comment (eric-wieser) [inline Mathlib/Logic/Lemmas.lean:79]: The naming guide says this should be called `not_injective`; though I'd be tempted to go further and prove `not_involutive` instead.

**Comment by urkud** [comment]
Should it go to `Logic/Lemmas` or `Logic/Function/Basic`? I'm not sure.

BTW, could you please add `not_bijective` and `not_surjective` as well?

**Comment by IvanRenison** [comment]
> Should it go to `Logic/Lemmas` or `Logic/Function/Basic`? I'm not sure.

I don't know, maybe yes. Any way, I don't know in witch part of `Logic/Function/Basic` should it be.
Tell me you what to do

**Comment by YaelDillies** [comment]
It should go in `Logic.Function.Basic`. You can add it wherever you want but I think just before the definition of `IsPartialInv` is best.

**Comment by IvanRenison** [comment]
> It should go in `Logic.Function.Basic`. You can add it wherever you want but I think just before the definition of `IsPartialInv` is best.

Before `IsPartialInv` it was not possible because `Involutive` is not defined yet. So I put it right after the namespace `Involutive`

File: Mathlib/Logic/Function/Basic.lean [hunk 2]
```diff
@@ -928,6 +928,14 @@ protected theorem eq_iff {x y : α} : f x = y ↔ x = f y :=
 
 end Involutive
 
+lemma not_involutive : Involutive Not := fun _ ↦ propext not_not
+
+lemma not_injective : Injective Not := not_involutive.injective
+
+lemma not_surjective : Surjective Not := not_involutive.surjective
+
+lemma not_bijective : Bijective Not := not_involutive.bijective
```
Comment (YaelDillies) [inline Mathlib/Logic/Function/Basic.lean:937]: You can remove line breaks when you have a sequence of similar one-line lemmas
```suggestion
lemma not_involutive : Involutive Not := fun _ ↦ propext not_not
lemma not_injective : Injective Not := not_involutive.injective
lemma not_surjective : Surjective Not := not_involutive.surjective
lemma not_bijective : Bijective Not := not_involutive.bijective
```

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**